stratum/CMakeLists.txt

```diff
@@ -350,58 +350,94 @@ target_link_libraries(stratum PUBLIC
     message(FATAL_ERROR "Cannot find library: tdi")
 endif()
 
+add_library(tdi SHARED IMPORTED)
+set_property(TARGET tdi PROPERTY IMPORTED_LOCATION ${LIBTDI})
+
+# tdi_json_parser
 find_library(LIBTDI_JSON_PARSER tdi_json_parser ${SDE_INSTALL_DIR}/lib)
 if(NOT LIBTDI_JSON_PARSER)
     message(FATAL_ERROR "Cannot find library: tdi_json_parser")
 endif()
 
+add_library(tdi_json_parser SHARED IMPORTED)
+set_property(TARGET tdi_json_parser
+             PROPERTY IMPORTED_LOCATION ${LIBTDI_JSON_PARSER})
+
+# target_sys
 find_library(LIBTARGET_SYS target_sys ${SDE_INSTALL_DIR}/lib)
 if(NOT LIBTARGET_SYS)
     message(FATAL_ERROR "Cannot find library: target_sys")
 endif()
 
+add_library(target_sys SHARED IMPORTED)
+set_property(TARGET target_sys PROPERTY IMPORTED_LOCATION ${LIBTARGET_SYS})
+
 if(TOFINO_TARGET)
+    # Tofino driver
     find_library(LIBDRIVER driver ${SDE_INSTALL_DIR}/lib)
     if(NOT LIBDRIVER)
         message(FATAL_ERROR "Cannot find library: driver")
     endif()
 
+    add_library(driver SHARED IMPORTED)
+    set_property(TARGET driver PROPERTY IMPORTED_LOCATION ${LIBDRIVER})
+
     target_link_libraries(stratum PUBLIC
-        ${LIBDRIVER}
-        ${LIBTDI}
-        ${LIBTDI_JSON_PARSER}
-        ${LIBTARGET_SYS}
+        driver
+        tdi
+        tdi_json_parser
+        target_sys
     )
 elseif(DPDK_TARGET)
+    # DPDK driver
     find_library(LIBDRIVER driver ${SDE_INSTALL_DIR}/lib)
     if(NOT LIBDRIVER)
         message(FATAL_ERROR "Cannot find library: driver")
     endif()
 
+    add_library(driver SHARED IMPORTED)
+    set_property(TARGET driver PROPERTY IMPORTED_LOCATION ${LIBDRIVER})
+
+    # bf_switchd_lib
     find_library(LIBBF_SWITCHD bf_switchd_lib ${SDE_INSTALL_DIR}/lib)
     if(NOT LIBBF_SWITCHD)
         message(FATAL_ERROR "Cannot find library: bf_switchd_lib")
     endif()
 
+    add_library(bf_switchd_lib SHARED IMPORTED)
+    set_property(TARGET bf_switchd_lib
+                 PROPERTY IMPORTED_LOCATION ${LIBBF_SWITCHD})
+
+    # target_utils
     find_library(LIBTARGET_UTILS target_utils ${SDE_INSTALL_DIR}/lib)
     if(NOT LIBTARGET_UTILS)
         message(FATAL_ERROR "Cannot find library: target_utils")
     endif()
 
+    add_library(target_utils SHARED IMPORTED)
+    set_property(TARGET target_utils
+                 PROPERTY IMPORTED_LOCATION ${LIBTARGET_UTILS})
+
     target_link_libraries(stratum PUBLIC
-        ${LIBDRIVER}
-        ${LIBBF_SWITCHD}
-        ${LIBTDI}
-        ${LIBTDI_JSON_PARSER}
-        ${LIBTARGET_UTILS}
-        ${LIBTARGET_SYS}
+        driver
+        bf_switchd_lib
+        tdi
+        tdi_json_parser
+        target_utils
+        target_sys
     )
 
-    target_link_directories(stratum PUBLIC ${DPDK_LIBRARY_DIRS})
-    target_link_options(stratum PUBLIC ${DPDK_LDFLAGS_OTHER})
+    target_link_directories(stratum PUBLIC
+        ${SDE_INSTALL}/lib
+        ${DPDK_LIBRARY_DIRS})
+
+    target_link_options(stratum PUBLIC ${DPDK_LD_FLAGS} ${DPDK_LDFLAGS_OTHER})
```

Comment:
What's difference between the two - DPDK_LD_FLAGS and DPDK_LDFLAGS_OTHER?

Reply:
-- DPDK_LIBRARY_DIRS="/home/dfoster/SDE/install/lib/x86_64-linux-gnu"
-- DPDK_LDFLAGS_OTHER="-Wl,--as-needed"
-- DPDK_LDFLAGS="-L/home/dfoster/SDE/install/lib/x86_64-linux-gnu;-Wl,--as-needed;-lrte_node;-lrte_graph;-lrte_flow_classify;-lrte_pipeline;-lrte_table;-lrte_pdump;-lrte_port;-lrte_fib;-lrte_ipsec;-lrte_vhost;-lrte_stack;-lrte_security;-lrte_sched;-lrte_reorder;-lrte_rib;-lrte_dmadev;-lrte_regexdev;-lrte_rawdev;-lrte_power;-lrte_pcapng;-lrte_member;-lrte_lpm;-lrte_latencystats;-lrte_kni;-lrte_jobstats;-lrte_ip_frag;-lrte_gso;-lrte_gro;-lrte_gpudev;-lrte_eventdev;-lrte_efd;-lrte_distributor;-lrte_cryptodev;-lrte_compressdev;-lrte_cfgfile;-lrte_bpf;-lrte_bitratestats;-lrte_bbdev;-lrte_acl;-lrte_timer;-lrte_hash;-lrte_metrics;-lrte_cmdline;-lrte_pci;-lrte_ethdev;-lrte_meter;-lrte_net;-lrte_mbuf;-lrte_mempool;-lrte_rcu;-lrte_ring;-lrte_eal;-lrte_telemetry;-lrte_kvargs;-lbsd"

which may mean that I really should have specified

`target_link_options(stratum PUBLIC ${DPDK_LDFLAGS} ${DPDK_LDFLAGS_OTHER})`